cvmfs/catalog.cc

```diff
@@ -63,6 +63,9 @@ Catalog::Catalog(const PathString &path,
   int retval = pthread_mutex_init(lock_, NULL);
   assert(retval == 0);
 
+  atomic_init32(&external_data_status_);
```

Comment:
It's always set in a locked context, but often read from an unlocked context.  You need to keep it an atomic to prevent partial reads.